hooks/src/index.js

```diff
@@ -239,8 +244,10 @@ function safeRaf(callback) {
 if (typeof window !== 'undefined') {
 	let prevRaf = options.requestAnimationFrame;
 	afterPaint = (component) => {
-		if ((!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) ||
-		    prevRaf !== options.requestAnimationFrame) {
+		if (
+			(!component._afterPaintQueued && (component._afterPaintQueued = afterPaintEffects.push(component)) === 1)
+			|| prevRaf !== options.requestAnimationFrame
```

Comment:
Hmm - I'm wondering how common the case of swapping `options.requestAnimationFrame` is that we'd need this. Was there a specific case where this was troublesome? Only thing I could think of was injecting a super long rAF and then wanting to cancel.

Reply:
I don't think its very common and personally would love to save some bytes here (-24 bytes!). It was orginally added in #1851 so perhaps @robertknight could chime and let us know his thoughts? Perhaps if we could fix the original `act` issue from #1851 in a different way, could we perhaps remove this check here?

Reply:
I am sure that the issue captured by the tests in #1851 should stay fixed, i.e. `act` needs to reliably flush any renders/effects triggered during its callback regardless of whether there are renders/effects happening outside of `act` callbacks or not. Tests can break in very confusing ways otherwise.

The Enzyme adapter installs a [`debounceRendering` hook](https://github.com/preactjs/enzyme-adapter-preact-pure/blob/67a70088f1aff9ddf0ef1df9e3a63e1e9c77f9aa/src/debounce-render-hook.ts#L21) to ensure that it can flush pending state updates when Enzyme asks for the current render output. It doesn't access `requestAnimationFrame` though, so if you can find an alternative way to keep the tests added in #1851 passing, that would be fine from my point of view.